e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -1168,14 +1171,14 @@ def _subset_time_series_dataset(self, ds: xr.Dataset, var: str) -> xr.Dataset:
         xr.Dataset
             The subsetted time series dataset.
         """
-        ds_sub = self._subset_vars_and_load(ds, var)
-
-        time_slice = self._get_time_slice(ds_sub)
-        ds_sub = ds_sub.sel(time=time_slice).squeeze()
+        time_slice = self._get_time_slice(ds)
+        ds_sub = ds.sel(time=time_slice).squeeze()
 
         if self.is_sub_monthly:
             ds_sub = self._exclude_sub_monthly_coord_spanning_year(ds_sub)
 
+        ds_sub = self._subset_vars_and_load(ds_sub, var)
+
```

Comment:
Just noting for future reference that we now time slice first then `.load()` the dataset in order to prevent memory allocation issues (e.g., dataset too large to load into memory before slicing).